src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -1158,31 +1100,43 @@ internal static StackValue[] GenerateRangeByAmount(decimal start, long amount, d
                 double.IsInfinity(endValue) || double.IsNaN(endValue) ||
                 svStep.IsDouble && (double.IsInfinity(svStep.RawDoubleValue) || double.IsNaN(svStep.RawDoubleValue)))
             {
-                runtimeCore.RuntimeStatus.LogWarning(
-                    WarningID.kInvalidArguments,
-                    Resources.kInvalidArgumentsInRangeExpression);
+                runtimeCore.RuntimeStatus.LogWarning(WarningID.kInvalidArguments, Resources.kInvalidArgumentsInRangeExpression);
                 return null;
             }
 
-            decimal start = new decimal(startValue);
-            decimal end = new decimal(endValue);
             bool isIntRange = svStart.IsInteger && svEnd.IsInteger;
 
             if (hasAmountOp)
             {
-                long amount = svEnd.ToInteger().opdata;
+                int amount = (int)svEnd.ToInteger().opdata;
                 if (amount < 0)
                 {
-                    runtimeCore.RuntimeStatus.LogWarning(
-                       WarningID.kInvalidArguments,
-                       Resources.kInvalidAmountInRangeExpression);
+                    runtimeCore.RuntimeStatus.LogWarning(WarningID.kInvalidArguments, Resources.kInvalidAmountInRangeExpression);
                     return null;
                 }
-                decimal stepsize = new decimal(svStep.ToDouble().RawDoubleValue);
-                return GenerateRangeByAmount(start, amount, stepsize, isIntRange);
+
+                if (amount == 0)
+                {
+                    return new StackValue[] { };
```

Comment:
Other code paths returned `null` instead, is there a difference between `new StackValue[] { }` and `null` for the callers of this method?